### PR TITLE
🛠️ Refactor: Split models to image and video

### PR DESCRIPTION
**Pull request by samet-akcay**
## Description

- Split anomalib models to image and video.

The following is the new structure after the refactor

```bash
src/anomalib/models
├── components
│   ├── base
│   ├── classification
│   ├── cluster
│   ├── dimensionality_reduction
│   ├── feature_extractors
│   ├── filters
│   ├── flow
│   ├── __init__.py
│   ├── layers
│   ├── sampling
│   └── stats
├── image
│   ├── cfa
│   ├── cflow
│   ├── csflow
│   ├── dfkde
│   ├── dfm
│   ├── draem
│   ├── efficient_ad
│   ├── fastflow
│   ├── ganomaly
│   ├── __init__.py
│   ├── padim
│   ├── patchcore
│   ├── README.md
│   ├── reverse_distillation
│   ├── rkde
│   └── stfpm
├── __init__.py
└── video
    ├── ai_vad
    └── __init__.py
```

### CODE REVIEW

File: src/anomalib/models/image/README.md
Comment (djdameln) [inline src/anomalib/models/image/README.md:1]: A similar README for video would be nice

Comment (samet-akcay) [reply src/anomalib/models/image/README.md:1]: I have now added a `README.md` for the video stuff as well.

**Review by djdameln — CHANGES_REQUESTED** [review]
1 minor comment

File: src/anomalib/models/video/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+<summary>Training the AiVad model on Avenue video dataset</summary>
+
+```python
+# Import the necessary modules
+from anomalib.data import Avenue
+from anomalib.models import AiVad
+from anomalib.engine import Engine
+
+# Load the avenue dataset, model and engine.
+datamodule = Avenue()
```
[the remark this answers is not in the document]
Comment (djdameln) [reply src/anomalib/models/video/README.md]: We could also make this the default behaviour of `Avenue`. If we decide this, then we should do the same for `ShanghaiTech` (`UCSDped` already has this as the default behaviour)

[the remark this answers is not in the document]
Comment (samet-akcay) [reply src/anomalib/models/video/README.md]: I think this would be a scope of another pr?

[the remark this answers is not in the document]
Comment (samet-akcay) [reply src/anomalib/models/video/README.md]: or is it just setting the default parameter to ?

**Review by djdameln — APPROVED** [review]

**Pull request merged**